### DISCUSSION TITLE
Also pass the event to the display method

### DISCUSSION THREAD

**Discussion by speich** [digitalfortress-tech/typeahead-standalone#24] [Ideas]
It would be helpful if the `display` method would also have the event passed as an argument (same as `onSubmit(event, selectedItem?)`). Then I could discern between click or keyup as well as backspace or other input.

**Comment by niketpathak** [comment]
Good idea. Pondering about it makes me realize that this could be used for tracking user behavior or for analytics too. Will launch a PR with the necessary updates soon. Thanks for the idea.

**Comment by niketpathak** [comment]
Hello @speich , this functionality has been implemented in [v4.4.0](https://www.npmjs.com/package/typeahead-standalone) deployed recently. The display callback now receives an optional 2nd parameter **event** that can be used as necessary

```js
display(selectedItem, event?: MouseEvent | KeyboardEvent | null) => {
    if (event) {
        // do something
    }
    return `${selectedItem.label}`;
}
```

It defaults to `null` and is left `undefined` when used for generating hints. 
Hope that helps!

**Comment by niketpathak** [comment]
As of [v4.21.0](https://www.npmjs.com/package/typeahead-standalone/v/4.21.0), there are two ways of doing this as described in the [FAQ](https://typeahead.digitalfortress.tech/#references?id=faq) section of the documents.
